src/modules/injectedWidget/hooks/useUpdateWidgetAppParams.ts

```diff
@@ -0,0 +1,20 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect } from 'react'
+
+import { InjectedWidgetParams, injectedWidgetParamsAtom } from '../state/injectedWidgetParamsAtom'
+
+const COW_SWAP_WIDGET_APP_UPDATE_KEY = 'cowSwapWidgetAppUpdate'
+
+export function useUpdateWidgetAppParams() {
+  const updateParams = useUpdateAtom(injectedWidgetParamsAtom)
+
+  useEffect(() => {
+    window.addEventListener('message', (event) => {
+      if (event.data.key !== COW_SWAP_WIDGET_APP_UPDATE_KEY) return
+
+      const appParams = event.data.params as InjectedWidgetParams
+
+      updateParams(appParams)
+    })
+  }, [updateParams])
+}
```

Comment:
nit: not sure if this belongs more in the updater dir than in the hooks. Given is very small, I think is fine to do it in the updater file itself

Fine leaving it here too (since its obviously a hook)

Reply:
Fixed, thank you!